WalletWasabi/Wallets/Wallet.cs

```diff
@@ -198,8 +200,9 @@ public void Logout()
 			ServiceConfiguration = Guard.NotNull(nameof(serviceConfiguration), serviceConfiguration);
 			FeeProvider = Guard.NotNull(nameof(feeProvider), feeProvider);
 
-			TransactionProcessor = new TransactionProcessor(BitcoinStore.TransactionStore, KeyManager, ServiceConfiguration.DustThreshold);
-			Coins = TransactionProcessor.Coins;
+			CoinsRegistry coinsRegistry = new(coinPrison);
+			TransactionProcessor = new TransactionProcessor(BitcoinStore.TransactionStore, KeyManager, ServiceConfiguration.DustThreshold, coinsRegistry);
+			Coins = coinsRegistry;
```

Comment:
No no, what Szpoti is doing here is okay because the `TransactionProcessor` should receive its dependencies. I mean, the Wallet should create the `CoinRegistry` and inject it into the `TransactionProcessor`  instead of the `TransactionProcessor` create the `CoinRegistry` and the wallet reference it.